### PR TITLE
Enforce semver version numbers in Manifest.json

### PR DESCRIPTION
**Pull request by cboulanger**
Since we are switching to qooxdoo versions compatible with the [semver standard](http://semver.org/spec/v2.0.0.html), we also need to adapt the validation of the qooxdoo verson required by a library in Manifest.json. At it is now, it allows „trunk“ and „master“ as versions, which is illegal in semver and doesn’t make any sense in any case since it is no indication of any time-independent identifiable version. This PR updates the regular expression used to validate the `info.qooxdoo-version` entry in the manifest json.

### CODE REVIEW

File: tool/pylib/generator/config/Manifest.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def patchLibEntry(self, libentry):
     @classmethod
     def schema_v1_0(self):
         patterns = {
-            "semver": r"^trunk$|master$|^\d+\.\d+(\.\d+)?(-[0-9]+-?)?([-a-zA-Z+][-a-zA-Z0-9.:-]*)?$",
+            # semver validation, see https://github.com/sindresorhus/semver-regex/blob/master/index.js
```
Comment (derrell) [inline tool/pylib/generator/config/Manifest.py]: The `^` and `$` that were in the deleted line are important to match the entire string.

**Review by derrell — CHANGES_REQUESTED** [review]

File: tool/pylib/generator/config/Manifest.py (hunk 1 above)
Comment (cboulanger) [reply tool/pylib/generator/config/Manifest.py]: Updated the rege accordingly...

**Comment by oetiker** [comment]
the problem of specifying semver compatible version ranges has been explored at length already ... so instead of inventing something new, I would would suggest to rely on existing solutions ... 

for example

https://git.daplie.com/coolaj86/semver-utils

to prevent problems with python generator I would simply leave its version compatibility thing as is and create a new key in the config

**Comment by cboulanger** [comment]
@oetiker This issue is not about semver **range**, but about enforcing the semver **version** it's already pretending to enforce (hence the name of the pattern in the code), but which it isn't because the pattern is not legal semver ( since it allows "trunk" and "master"). Does this address your worries?

**Comment by oetiker** [comment]
for now I have specified something like 5.0 or 5.0.x in my contribs to point out that they would work with all 5.0ish versions ... both would not be semver compatible ... 

master and trunk are neither ... true

but instead of fixing this half way ... how about allowing contribs to specify a semver range

**Comment by cboulanger** [comment]
@oetiker that's the idea, but in a different issue- it will be done in a different config key which is picked up by new compiler. So all you'll
Have to do is to fix the qooxdoo-version value and add a range in the new config-key, tentatively called 'qooxdoo-range'. I will create a separate issue for this..

**Comment by oetiker** [comment]
great

**Review by oetiker — APPROVED** [review]

**Review by derrell — APPROVED** [review]
LGTM

**Review by derrell — APPROVED** [review]
I don't understand the two similar-looking (identical) PRs

**Review by cajus — APPROVED** [review]

**Review by johnspackman — APPROVED** [review]

**Review by patrick-fischer-hirschstein — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Comment by level420** [comment]
6 approvals, merging.
thank you @cboulanger

**Pull request merged**